src/util/shell.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/shell.h>
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#ifdef ENABLE_EXTERNAL_SIGNER
+#if defined(__GNUC__)
+// Boost 1.78 requires the following workaround.
+// See: https://github.com/boostorg/process/issues/235
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnarrowing"
+#endif
+#include <boost/process.hpp>
+#include <memory>
+#include <ostream>
+#include <vector>
+#if defined(__GNUC__)
+#pragma GCC diagnostic pop
+#endif
+#endif // ENABLE_EXTERNAL_SIGNER
+
+#include <logging.h>
+#include <tinyformat.h>
+#include <univalue.h>
+#include <util/string.h>
+
+#include <cstdlib>
+#include <stdexcept>
+#include <string>
+
+#ifdef WIN32
+#include <compat/compat.h>
```

Comment:
In e5243d3676136e4b0a590e76a07c86e62ca7d1f7:
If this is for `windows.h`, would rather just include that.